python/lsst/ts/scriptqueue/ui/request_cmd.py

```diff
@@ -0,0 +1,385 @@
+                                       help="Monitor script execution all the way to the end.")
+        self.__run_parser.add_argument("--timeout", dest="timeout", default=120., type=float,
+                                       help="Timeout for monitoring scripts.")
+
+        self.__set_log_parser = argparse.ArgumentParser(prog="log_parser")
+        self.__set_log_parser.add_argument("-v", "--verbose", dest="verbose",
+                                           action='count', default=0,
+                                           help="Set the verbosity for the console logging.")
+
+        self.__set_script_log_parser = argparse.ArgumentParser(prog="script_log")
+        self.__set_script_log_parser.add_argument("-v", "--verbose", dest="verbose",
+                                                  action='count',
+                                                  default=0,
+                                                  help="Set the verbosity for the console logging.")
+        self.__set_script_log_parser.add_argument("-i", "--index",
+                                                  type=int,
+                                                  help="SalIndex of script to change log level.")
+
+    def do_heartbeat(self, arg):
+        """Listen for queue heartbeat."""
+        try:
+            nbeats = 10
+            try:
+                nbeats = int(arg)
+            except ValueError as e:
+                self.log.debug(f"Using default heartbeat number {nbeats}")
+            for i in range(nbeats):
+                self.model.listen_heartbeat()
+                self.log.info(f"Heartbeat {i+1}/{nbeats}...")
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_enable(self, arg):
+        """Enable the queue."""
+        try:
+            self.model.enable_queue()
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_list(self, arg):
+        """List available scripts."""
+        try:
+            print("Listing all available scripts.")
+
+            script_list = self.model.list()
+
+            print('List of external scripts:')
+            for script in script_list["external"]:
+                print(f'\t - {script}', )
+
+            print('List of standard scripts:')
+            for script in script_list["standard"]:
+                print(f'\t - {script}', )
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_pause(self, arg):
+        """Pause the queue."""
+        print("Pausing queue.")
+        try:
+            self.model.pause_queue()
+        except Exception as e:
+            self.log.exception(e)
+
+        return False
+
+    def do_resume(self, arg):
+        """Resume queue."""
+        print("Resuming queue.")
+        try:
+            self.model.resume_queue()
+        except Exception as e:
+            self.log.exception(e)
```

Comment:
You can ditch all these try/except blocks in do_x methods by adding this method to your class:
```
    def onecmd(self, *args):
        try:
            return super().onecmd(*args)
        except Exception as e:
            self.log.exception(e)
```